test/index.js

```diff
@@ -152,7 +152,14 @@ describe('maybe', function() {
 
     it('provides an "extend" method', function() {
       eq(S.Nothing().extend.length, 1);
-      eq(S.Nothing().extend(function() { return 42; }), S.Nothing());
+      eq(S.Nothing().extend(function(x) { return x.value; }), S.Nothing());
+
+      // Extend Property Test
+      var w = S.Nothing();
+      var f = function(x) { return x.value + 1; };
+      var g = function(x) { return x.value * x.value; };
+      eq(w.extend(g).extend(f),
+         w.extend(function(_w) { return f(_w.extend(g)); }));
```

Comment:
This is the associativity law, which can be written `extend f ∘ extend g = extend (f ∘ extend g)`.

Let's update the comment to reflect this (we use `// associativity` elsewhere).